src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Microsoft.DotNet.ServiceFabric.ServiceHost;
+using Microsoft.Extensions.Logging;
+using Microsoft.ServiceFabric.Data;
+using Microsoft.ServiceFabric.Data.Collections;
+using Microsoft.Dotnet.GitHub.Authentication;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.DotNet.Git.IssueManager;
+using System.Text;
+using Newtonsoft.Json.Linq;
+using System.Net.Http;
+using Microsoft.TeamFoundation.Common;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace DependencyUpdateErrorProcessor
+{
+    /// <summary>
+    /// An instance of this class is created for each service replica by the Service Fabric runtime.
+    /// </summary>
+    internal sealed class DependencyUpdateErrorProcessor : IServiceImplementation
+    {
+        public DependencyUpdateErrorProcessor(
+            IReliableStateManager stateManager,
+            ILogger<DependencyUpdateErrorProcessor> logger,
+            BuildAssetRegistryContext context,
+            IOptions<DependencyUpdateErrorProcessorOptions> dependencyUpdateErrorProcessorOptions
+            )
+        {
+            StateManager = stateManager;
+            Logger = logger;
+            Context = context;
+            DependencyUpdateErrorProcessorOptions = dependencyUpdateErrorProcessorOptions.Value;
```

Comment:
@alexperovich mentioned that is also a way to get the values from DependencyUpdateErrorProcessorOptions class, that way I don't have to change the type to IOptions.

Reply:
I think you are supposed to store the IOptions (or, if the options change over time, which these will, because there are keys in them, an IOptionsMonitor, so you can react to value changes).